src/main/java/io/codelex/flightplanner/flights/FlightRepository.java

```diff
@@ -0,0 +1,90 @@
+package io.codelex.flightplanner.flights;
+
+import io.codelex.flightplanner.customerapi.PageResult;
+import io.codelex.flightplanner.customerapi.SearchFlightsRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Repository
+public class FlightRepository {
+
+    private List<Flight> allFlights = new ArrayList<>();
+    private Long counter = 0L;
+
+    public Long getId() {
+        return counter++;
+    }
+
+    public void clear() {
+        allFlights.clear();
+    }
+
+    public Flight addFlight(Flight addFlight) throws ResponseStatusException {
+        for (Flight flight : allFlights) {
+            if (flight.isSameFlight(addFlight)) {
+                throw new ResponseStatusException(HttpStatus.CONFLICT, "This flight already exists");
+            }
+        }
+        allFlights.add(addFlight);
+        return addFlight;
+    }
+
+    public Flight fetchFlight(Long id) {
+        for (Flight flight : allFlights) {
+            if (flight.getId() == id) {
+                return flight;
+            }
+        }
+        throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+    }
+
+    public PageResult<Flight> searchFlights(SearchFlightsRequest searchFlightsRequest) {
+        final int RESULTS_PER_PAGE = 20;
+        PageResult<Flight> pageResult = new PageResult<>();
+
+        List<Flight> flightList = filterResults(allFlights, searchFlightsRequest);
+        pageResult.setItems(flightList);
+
+        pageResult.setPage((int) Math.ceil((double) flightList.size() / RESULTS_PER_PAGE));
+        pageResult.setTotalItems(flightList.size());
+
+        return pageResult;
+    }
+
+    private List<Flight> filterResults(List<Flight> list, SearchFlightsRequest searchFlightsRequest) {
+        LocalDate localDate = LocalDate.parse(searchFlightsRequest.getDepartureDate());
+        return list.stream()
+                .filter(flight -> flight.getFrom().getAirport().equalsIgnoreCase(searchFlightsRequest.getFrom()))
+                .filter(flight -> flight.getTo().getAirport().equalsIgnoreCase(searchFlightsRequest.getTo()))
+                .filter(flight -> flight.getDepartureTime().toLocalDate().equals(localDate))
+                .collect(Collectors.toList());
+    }
+
+    public void deleteFlight(Long id) {
+        allFlights.removeIf(flight -> flight.getId() == id);
+    }
+
+    public List<Airport> searchByAirport(String airportSearch) {
```

Comment:
Šo varbūt vajadzētu pārsaukt, jo searchByAirport - ar to drīzāk saprastu, ka tiek meklēts pēc lidostas. Iedomājies kodu: `flightRepository.searchByAirport("xxx")`  - te 100% var saprast, ka metodei būtu jāatgriež flights, kuri tiek meklēti pēc lidostas. Drīzāk varētu saukties vienkārši `searchAirports`